### DISCUSSION TITLE
Automatic reload failing: OSError: [WinError 87] The parameter is incorrect

### DISCUSSION THREAD

**Discussion by BramVanroy** [fastapi/fastapi#7835] [Questions]
### Describe the bug

Hi there. After all the enthusiastic comments on Twitter, I couldn't wait until tomorrow to check FastAPI out. And _it's great_! Really impressive work!

I am just testing the API itself for now (not integrated with front-end) so I do my tests in the Swagger UI which is incredibly useful. However, I found that each time I change something in my code the server tries to reload (using `--reload`) but it fails. It thinks that it is successful, but reloading the page results in a page that just keeps loading. As you can see below, it does throw an error with a Traceback, but nevertheless continues and thinks it did a good job.

```
[32mINFO[0m:     Started server process [[36m16924[0m]
[31mERROR[0m:    Accept failed on a socket
socket: <socket.socket fd=476, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('127.0.0.1', 8000)>
Traceback (most recent call last):
  File "c:\users\bramv\appdata\local\programs\python\python37\Lib\asyncio\proactor_events.py", line 668, in loop
    f = self._proactor.accept(sock)
  File "c:\users\bramv\appdata\local\programs\python\python37\Lib\asyncio\windows_events.py", line 489, in accept
    self._register_with_iocp(listener)
  File "c:\users\bramv\appdata\local\programs\python\python37\Lib\asyncio\windows_events.py", line 650, in _register_with_iocp
    _overlapped.CreateIoCompletionPort(obj.fileno(), self._iocp, 0, 0)
OSError: [WinError 87] The parameter is incorrect
[32mINFO[0m:     Waiting for application startup.
[32mINFO[0m:     Application startup complete.
```

### To Reproduce

Steps to reproduce the behavior with a minimum self-contained file.

Replace each part with your own scenario:

1. Create a file with:

```Python
from fastapi import FastAPI

app = FastAPI()


@app.get("/")
def read_root():
    return {"Hello": "World"}
```

3. Open the browser and call the endpoint `/`.
4. It returns a JSON with `{"Hello": "World"}`.
5. Change something in the source code. The server will try to reload.
6. Try to refresh the page. It will keep loading but never succeed.

### Environment

- OS: Windows 10 x64
- FastAPI version: 0.44.0
- Python version: 3.7.5

### Additional context

It is possible that I am doing something wrong since this is the first time me using this. But I wouldn't know what! I change something in the code, expect the server to reload. It tries to but it fails, and still keeps going thinking it did alright. That being said, it is also possible that this is one of those Windows-specific bugs as indicated by this line "OSError: [WinError 87] The parameter is incorrect".

**Comment by euri10** [comment]
You should look at the uvicorn repository for this kind of issues :wink:

**Comment by prostomarkeloff** [comment]
This is windows local problem. You can create issue in uvicorn repo

**Comment by BramVanroy** [comment]
I found the (possibly incorrectly closed) issue [here](https://github.com/encode/uvicorn/issues/477), so I'll continue the discussion there.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :clap: :bow: 

Thanks for reporting back and closing the issue @BramVanroy  :+1:
